pkg/webhook/cluster/validation/validation.go

```diff
@@ -247,6 +240,50 @@ func validateClusterNetworkingConfigUpdateImmutability(c, oldC *kubermaticv1.Clu
 	return allErrs
 }
 
+func validateCNIUpdate(cni *kubermaticv1.CNIPluginSettings, oldCni *kubermaticv1.CNIPluginSettings, labels map[string]string) field.ErrorList {
+	specFldPath := field.NewPath("spec")
+
+	if cni == nil && oldCni == nil {
+		return field.ErrorList{} // allowed for backward compatibility with older KKP with existing clusters with no CNI settings
+	}
+	if oldCni != nil && cni == nil {
+		return field.ErrorList{field.Required(specFldPath.Child("cniPlugin"), "CNI plugin settings cannot be removed")}
+	}
+	if oldCni == nil && cni != nil {
+		if _, ok := labels[unsafeCNIUpgradeLabel]; ok {
+			return field.ErrorList{} // allowed for migration path from older KKP with existing clusters with no CNI settings
+		}
+		return field.ErrorList{field.Forbidden(specFldPath.Child("cniPlugin"),
+			fmt.Sprintf("cannot add CNI plugin settings, unless %s label is present", unsafeCNIUpgradeLabel))}
+	}
+	if cni.Type != oldCni.Type {
+		if _, ok := labels[unsafeCNIMigrationLabel]; ok {
+			return field.ErrorList{} // allowed for CNI type migration path
+		}
+		return field.ErrorList{field.Forbidden(specFldPath.Child("cniPlugin", "type"),
+			fmt.Sprintf("cannot change CNI plugin type, unless %s label is present", unsafeCNIMigrationLabel))}
+	}
+	if cni.Version != oldCni.Version {
+		newV, err := semver.NewVersion(cni.Version)
+		if err != nil {
+			return field.ErrorList{field.Invalid(specFldPath.Child("cniPlugin", "version"), cni.Version,
+				fmt.Sprintf("couldn't parse CNI version `%s`: %v", cni.Version, err))}
+		}
+		oldV, err := semver.NewVersion(oldCni.Version)
+		if err != nil {
+			return field.ErrorList{field.Invalid(specFldPath.Child("cniPlugin", "version"), oldCni.Version,
+				fmt.Sprintf("couldn't parse CNI version `%s`: %v", oldCni.Version, err))}
+		}
```

Comment:
This allows upgrade from 1.2.0 to 5.3.0. 

The check for major version match is missing. :thinking:

Reply:
indeed, thx :)

Reply:
A test for this would be good too so that we never regress.